camel/functions/openai_function.py

```diff
@@ -107,7 +108,9 @@ def _create_mol(name, field):
         return create_model(name, **field)
 
     model = _create_mol(to_pascal(func.__name__), fields)
-    parameters_dict = model.model_json_schema()
+    # .schema() is deprecated in pydantic v2.
+    # the result would be identical to .model_json_schema() in v2
+    parameters_dict = model.schema()
```

Comment:
Why not use `PYDANTIC_V2` here?

Reply:
> Why not use `PYDANTIC_V2` here?

For v2 users calling the deprecated `.schema` will call `.model_json_schema`. https://github.com/pydantic/pydantic/blob/e131d53a07da080c2ba642775efa9e81cb2cf0ba/pydantic/main.py#L1261-L1268. So I just call `.schema`. But well we can add an if case to avoid the warning msg for v2 users.

Reply:
Sounds good. Could you please add it here?